dsc/workflows/opencourseware.py

```diff
@@ -0,0 +1,183 @@
+import json
+import logging
+import zipfile
+from collections.abc import Iterator
+from typing import Any
+
+import smart_open
+
+from dsc.exceptions import ReconcileError
+from dsc.utilities.aws.s3 import S3Client
+from dsc.workflows.base import Workflow
+
+logger = logging.getLogger(__name__)
+
+
+class OpenCourseWare(Workflow):
+    """Workflow for OpenCourseWare (OCW) deposits.
+
+    The deposits managed by this workflow are requested by the
+    Scholarly Communications and Collections Strategy (SCCS) department
+    and were previously deposited into DSpace@MIT by Technical services staff.
+    """
+
+    workflow_name: str = "opencourseware"
+    metadata_mapping_path: str = "dsc/workflows/metadata_mapping/opencourseware.json"
+
+    def reconcile_bitstreams_and_metadata(self) -> None:
+        """Reconcile bitstreams against item metadata.
+
+        Generate a list of bitstreams without item metadata.
+
+        For OpenCourseWare deposits, the zip files are the bitstreams to be deposited
+        into DSpace, but they also must contain a 'data.json' file, representing the
+        metadata. As such, the 'reconcile' method only determines whether there are any
+        bitstreams without metadata (any zip files without a 'data.json').
+        Metadata without bitstreams is not calculated as for a 'data.json' file to
+        exist, the zip file must also exist.
+        """
+        item_identifiers = []
+        bitstreams_without_metadata = []
+        s3_client = S3Client()
+        for file in s3_client.files_iter(
+            bucket=self.s3_bucket, prefix=self.batch_path, file_type=".zip"
+        ):
+            zip_file = f"s3://{self.s3_bucket}/{file}"
+            item_identifier = file.split("/")[-1].removesuffix(".zip")
```

Comment:
Note: This function contains similar steps to `OpenCourseWare.item_metadata_iter`; only, instead of yielding a dictionary of metadata, it checks whether metadata can be extracted from the zip file--tracking item identifiers in two lists:
1. `item_identifiers`: list of item identifiers for all zip files
2. `bitstreams_without_metadata`: list of item identifiers for all zip files without a 'data.json' file.

Reply:
@ghukill Did you mean these 2 lines? If so, I agree

```
zip_file = f"s3://{self.s3_bucket}/{file}"
item_identifier = file.split("/")[-1].removesuffix(".zip")
```

Reply:
Let me know if [these changes](https://github.com/MITLibraries/dspace-submission-composer/pull/91/commits/3ca48aa313858b55eb1e0474db03cc268abb91c7) are in line with what you were thinking, @ghukill and @ehanson8 !